scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/LiveJobs.java

```diff
@@ -637,6 +635,8 @@ public TerminationData taskTerminatedWithResult(TaskId taskId, TaskResultImpl re
                         !onErrorPolicyInterpreter.requiresPauseJobOnError(task) &&
                         !onErrorPolicyInterpreter.requiresCancelJobOnError(task)) {
                         jobData.job.increaseNumberOfFaultyTasks(taskId);
+                        // remove the parent tasks results if task fails and job is canceled
+                        task.removeParentTasksResults();
```

Comment:
Why we do this?

Reply:
We want to release the memory asap to avoid a possible memory overflow (propagated variables can be too big)